classes/Event.php

```diff
@@ -542,7 +542,7 @@ protected function notifyOnRsvp(string $type, int $to = null): void {
 					}
 				}
 	
-				if ($this->register_nologin) {
+				if ($this->register_nologin && !elgg_is_logged_in()) {
```

Comment:
can you change this to `$to_entity instanceof \EventRegistration` as that will make sure this function works always as expected (even if used logged out or logged in)

Reply:
You are right, that's a better implementation: done

Reply:
Thanks for updating... it would help next time to squash your commits (if they apply to the same change). Commit messages will end up in the release notes. If makes no sense to have 2 messages related to the same change. I have squashed the commits for you this time.

Reply:
ok, I will do it next time